### PR TITLE
Clean MUI imports

### PR DESCRIPTION
**Pull request by flomillot**
Code cleaning on imports that was initialy done on #352 because of issues with default imports.

We use only named imports from first level for MUI https://github.com/mui/material-ui/issues/30671#issuecomment-1613525947
It's strongly advised to only use named imports in libraries, cf https://esbuild.github.io/content-types/#default-interop

> If you are a library author: When writing new code, you should strongly consider avoiding the default export entirely. It has unfortunately been tainted with compatibility problems and using it will likely cause problems for your users at some point.

### CODE REVIEW

File: vite.config.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
+import react from '@vitejs/plugin-react';
```
Comment (sBouzols) [inline vite.config.ts:1]: typescript file -> license header

File: package.json [hunk 2]
```diff
@@ -1,27 +1,23 @@
 {
     "name": "@gridsuite/commons-ui",
-    "version": "0.43.0",
+    "version": "0.44.0",
```
Comment (klesaulnier) [inline package.json:3]: should we update commons-ui version in a specific PR as usual ?

File: src/components/TopBar/TopBar.jsx [hunk 3]
```diff
@@ -40,8 +40,7 @@ import {
     Settings as SettingsIcon,
     WbSunny as WbSunnyIcon,
 } from '@mui/icons-material';
-import { darken } from '@mui/material/styles';
-import { styled } from '@mui/system';
+import { darken, styled } from '@mui/system';
```
Comment (klesaulnier) [inline src/components/TopBar/TopBar.jsx:43]:
```suggestion
import { styled } from '@mui/system';
import { darken } from '@mui/material';
```

File: tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,16 @@
+{
+  "compilerOptions": {
+    "module": "ESNext",
```
Comment (klesaulnier) [inline tsconfig.json:3]:
```suggestion
    "module": "esnext",
```

File: package.json (hunk 2 above)
Comment (flomillot) [reply package.json:3]: Yes mb

File: src/components/TopBar/TopBar.jsx (hunk 3 above)
Comment (flomillot) [reply src/components/TopBar/TopBar.jsx:43]: It's actually the same

File: tsconfig.json (hunk 4 above)
Comment (flomillot) [reply tsconfig.json]: Done

File: vite.config.ts (hunk 1 above)
Comment (flomillot) [reply vite.config.ts:8]: Done

File: jest.config.ts [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import type { Config } from 'jest';
```
[the remark this answers is not in the document]
Comment (flomillot) [reply jest.config.ts:8]: Done

**Review by klesaulnier — APPROVED** [review]

**Comment by Tristan-WorkGH** [comment]
@flomillot I've resolved the `legacy-peer-deps` in #350

**Comment by Tristan-WorkGH** [comment]
@flomillot I think I've fixed the external dependencies with #351

**Review by Tristan-WorkGH — APPROVED** [review]

**Pull request merged**